src/NationalInstruments.Analyzers/Correctness/ApprovedNamespaceCodeFixProvider.cs

```diff
@@ -33,18 +37,55 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 var namespaceName = (await location.SourceTree.GetRootAsync(context.CancellationToken).ConfigureAwait(false))
                     .FindNode(location.SourceSpan)
                     .ToString();
-                var codeAction = CodeAction.Create(
-                    Resources.NI1800_CodeFixTitle,
-                    cancellationToken => AddNamespaceAsync(context, namespaceName),
-                    nameof(ApprovedNamespaceCodeFixProvider));
-                context.RegisterCodeFix(codeAction, context.Diagnostics);
+                var approvedNamespacesFilePaths = diagnostic.Properties.Values;
+                foreach (var path in approvedNamespacesFilePaths)
+                {
+                    context.RegisterCodeFix(new ApprovedNamespaceCodeAction(context, namespaceName, path), context.Diagnostics);
+                }
             }
         }
 
-        private Task<Document> AddNamespaceAsync(CodeFixContext context, string namespaceName)
+        private class ApprovedNamespaceCodeAction : CodeAction
         {
-            ApprovedNamespaceAnalyzer.ApproveNamespace(namespaceName);
-            return Task.FromResult(context.Document);
+            private readonly string _namespaceName;
+            private readonly CodeFixContext _context;
```

Comment:
Would be nice if order followed constructor order

Reply:
Done